src/smart-components/group/add-group/users-list.js

```diff
@@ -1,82 +1,96 @@
+              ),
+              { title: userLinks ? <Link to={`/users/detail/${username}`}>{username.toString()}</Link> : username.toString() },
+              email.toString(),
+              firstName.toString(),
+              lastName.toString(),
+              {
+                title: (
+                  <Label key="status" color={isActive && 'green'}>
+                    {intl.formatMessage(isActive ? messages.active : messages.inactive)}
+                  </Label>
                 ),
-                { title: userLinks ? <Link to={`/users/detail/${username}`}>{username.toString()}</Link> : username.toString() },
-                email.toString(),
-                firstName.toString(),
-                lastName.toString(),
-                {
-                  title: (
-                    <Label key="status" color={isActive && 'green'}>
-                      {intl.formatMessage(isActive ? messages.active : messages.inactive)}
-                    </Label>
-                  ),
-                  props: {
-                    'data-is-active': isActive,
-                  },
+                props: {
+                  'data-is-active': isActive,
                 },
-              ],
-              selected: Boolean(checkedRows && checkedRows.find((row) => row.uuid === username)),
-            },
-          ],
-          []
-        )
-      : [];
-  };
+              },
+            ],
+            selected: Boolean(checkedRows && checkedRows.find((row) => row.uuid === username)),
+          },
+        ],
+        []
+      )
+    : [];
+};
 
-const UsersList = ({ users, fetchUsers, updateUsersFilters, isLoading, pagination, selectedUsers, setSelectedUsers, userLinks, inModal, props }) => {
+const UsersList = ({ selectedUsers, setSelectedUsers, userLinks, inModal, props }) => {
+  const intl = useIntl();
+  const history = useHistory();
+  const dispatch = useDispatch();
   const defaultPagination = useSelector(({ userReducer: { users } }) => ({
     limit: inModal ? users.meta.limit : users.pagination.limit || defaultSettings.limit,
     offset: inModal ? users.meta.offset : users.pagination.offset || defaultSettings.offset,
     count: inModal ? users.meta.count : users.pagination.count,
     redirected: !inModal && users.pagination.redirected,
   }));
-  const intl = useIntl();
-  const history = useHistory();
+  const selector = ({ userReducer: { users, isUserDataLoading } }) => ({
+    users: users.data && users.data.map((data) => ({ ...data, uuid: data.username })),
+    pagination: users.meta,
+    isLoading: isUserDataLoading,
+  });
+  const { users, pagination, isLoading } = useSelector(selector);
 
+  let stateFilters = useSelector(
+    ({
+      userReducer: {
+        users: { filters },
+      },
+    }) => (history.location.search.length > 0 || Object.keys(filters).length > 0 ? filters : { status: ['Active'] })
+  );
+
+  const fetchData = (apiProps) => {
+    return dispatch(fetchUsers(apiProps));
+  };
```

Comment:
Since this won't be changed during re-renders I would change this to a `useCallback`
```suggestion
  const fetchData = useCallback((apiProps) => {
    return dispatch(fetchUsers(apiProps));
  }, [dispatch]);
```